src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.cs

```diff
@@ -182,6 +185,67 @@ public void GarbageCollect(Func<string, IWorkloadResolver> getResolverForWorkloa
             }
         }
 
+        public string InstallWorkloadSet(string path)
+        {
+            return ModifyWorkloadSet(path, InstallAction.Install);
+        }
+
+        public void RollBackWorkloadSetInstallation()
+        {
+            if (_newWorkloadSetPath is not null)
```

Comment:
This should probably check for ! nullorwhitespace since an empty string wouldn't make any sense here I think. Also, seems odd to record an internal state in a private member called `_newWorkloadSetPath`. It's only new at the time of being set, but then it's the actual value. Can we parameterize it at all?

Reply:
I'd argue that if it's white space (and not null), we have a bug anyway, and we may as well try to go forward and crash so we at least find out afterwards.

What do you mean by parameterizing it? Like make it RollbackWorkloadSetInstallation(string) then pass the workload set path to that?

Reply:
yup, add a string parameter is what I meant.